state/builder.go

```diff
@@ -26,115 +27,153 @@ func buildKong(kongState *KongState, raw *utils.KongRawState) error {
 			return fmt.Errorf("inserting service into state: %w", err)
 		}
 	}
+	ensureService := func(serviceID string) (bool, *kong.Service, error) {
+		s, err := kongState.Services.Get(serviceID)
+		if err != nil {
+			if err == ErrNotFound {
+				return false, nil, nil
+			}
+			return false, nil, fmt.Errorf("looking up service %q: %w", serviceID, err)
+
+		}
+		return true, utils.AsKongService(s.Service), nil
+	}
```

Comment:
Is there a reason these `ensureFoo()` functions are inlined inside `buildKong()?` It doesn't look like they use anything from `buildKong()`'s dynamic scope, so they should be extracted into independent functions. No functional change, but it'll make `buildKong()` more readable.

Reply:
They actually share `kongState` which is changed across the function, but it can definitely be passed around.

Reply:
https://github.com/Kong/deck/pull/662/commits/cdf81cd83d6308955b89a3290069eb117f24c14c